[PHYW-527] Libraries in CMake

serranozack opened the pull request.

serranozack committed:
otdoa: handle library import in global cmake

Pulls in OTDOA library from ext/lib/bin/otdoa

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>

serranozack committed:
otdoa: control filesystem settings through Kconfig

Replaces LittleFS and External Storage options in
per-project CMakeLists.txt files

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>

serranozack committed:
otdoa: support linking dev build of OTDOA

aligns Kconfig with changes in OTDOA library

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>

serranozack committed:
otdoa: fix failing compliance check

Compliance failed because of incorrect leading
whitespace in OTDOA AL Kconfig

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>

serranozack committed:
otdoa: update overview doc

Replaces reference to CONFIG_OTDOA_DEBUG_LOGS with list of OTDOA
library type Kconfig choices

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>

nriedel reviewed (changes requested):
`I can't build the `otdoa_obfs_lib` target.  It fails with this:
```
-- Found gen_kobject_list: /home/nriedel/ncs-main/zephyr/scripts/build/gen_kobject_list.py
OBFUSCATE: On
LIB PROD: 
LIB  DBG: 
LIB  DEV: 
CMake Error at CMakeLists.txt:30 (message):
  No otdoa build type specified
```

nriedel commented on lib/otdoa_al/Kconfig:
```diff
@@ -29,6 +26,27 @@ config OTDOA_SHELL_COMMANDS
+		bool "Developer release with capture functions"
+	endchoice
+
+config OTDOA_LITTLE_FS
```
I get this warning when building the sample app:
```
/home/nriedel/ncs-main/nrf/samples/cellular/otdoa/src/main.c:21:18: warning: implicit declaration of function 'mount_little_fs'; did you mean 'mount_fat_fs'? [-Wimplicit-function-declaration]
   21 |         if (0 == mount_little_fs()) {
      |                  ^~~~~~~~~~~~~~~
      |                  mount_fat_fs
```
I think this means that `CONFIG_OTDOA_LITTLE_FS` is not actually defined for this build.

I think we should modify `otdoa_fs.h` to just use the existing `HOST` define here:
```
/* NORDIC uses Zephyr RTOS FS abstraction layer */
#if CONFIG_OTDOA_LITTLE_FS
```
Then `CONFIG_OTDOA_LITTLE_FS` can just define if we include the functions in `otdoa_little_fs.c` in the AL.  If we don't then the user has to supply those functions (e.g. using a FAT FS or something similar).

Does that all make sense?

nriedel replied:
@serranozack I think we still need to update CONFIG_OTDOA_LITTLE_FS, right?

nriedel replied:
So my question is, is this done?
Then CONFIG_OTDOA_LITTLE_FS can just define if we include the functions in otdoa_little_fs.c in the AL. If we don't then the user has to supply those functions (e.g. using a FAT FS or something similar).

serranozack replied:
Do you mean updating the Kconfig description?

nriedel replied:
No, I mean making CONFIG_OTDOA_LITTLE_FS a real kconfig parameter.

nriedel reviewed (changes requested):
Good stuff!  It builds fine for me now.  I think there is a little cleanup we should do around the little FS config as described below.

serranozack committed:
otdoa: fix warning when building sample

Compile gate in the littlefs header had the conditional
the wrong way around.

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>

serranozack committed:
otdoa: move otdoa_little_fs.c from sample to al

draft commit, there's an include issue with lfs.h

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>

nriedel committed:
lib: otdoa_al:  Fix issue with littlefs module include path

Explicitly added modules/fs/littlefs to the include path.

Signed-off-by: Neal Riedel <nriedel@serranosystems.com>

nriedel committed:
lib: otdoa_al: Fix little FS compilation issue

Conditionally add otdoa_little_fs.c to otdoa_al only when
CONFIG_OTDOA_LITTLE_FS is set.

Signed-off-by: Neal Riedel <nriedel@serranosystems.com>
